### DISCUSSION TITLE
How to format object creation syntax?

### DISCUSSION THREAD

**Discussion by vrrao01** [dotnet/roslyn#62353] [Q&A] [answered]
I'm using Roslyn to add a few `ObjectCreationExpressionSyntax` nodes. Then I format the document using `Formatter.Format()`. However, all the arguments appear on a single line making it hard to read. For example:  
The added line looks like 
`new CutomGroup("groupName", GroupCategories.Default, new Item[] { ShopItem.Instance("name"), DefaultItem.Instance("name2"), DiscountItem.GetInstance("name3", 20) });`
I would like each of the arguments in the constructor and array initializer to appear on a new line with proper indentation wrt to the rest of the document. Is there any way to achieve this?

**Comment by CyrusNajmabadi** [comment, marked as answer]
The formatting engine doesn't support formatting initializer contents.  If you want this formatted you will have to control the trivia on these tokens by hand. Sorry!

**Reply by vrrao01** [reply]
Ohh I see. Thank you. Also, I’m quite new to Roslyn and it’d really help if I get an example that shows how to control the trivia manually. 
I’ve tried adding NewLines after the comma token,etc but it doesn’t align. So I guess I’ll also have to keep track of the identation level right ?

**Reply by CyrusNajmabadi** [reply]
That's correct!
